backend/src/api/open_source_project.ts

```diff
@@ -0,0 +1,263 @@
+  
+  return {
+    
+    statusCode: 200,
+    body: JSON.stringify(project),
+  };
+});
+
+
+/**
+ * @swagger
+ *
+ * /project/listOrgs/{orgId}:
+ *  list:
+ *    description: List open source projects for given organization.
+ *    parameters:
+ *      - in: path
+ *        name: orgId
+ *        required: true
+ *        description: Organization ID
+ *    tags:
+ *    - Organizations
+ */
+export const listByOrg = wrapHandler(async (event) => {
+  console.log("im here");
+  const orgId = event.pathParameters?.orgId;
+  // check permissions
+  if (!orgId || (!isGlobalViewAdmin(event) && !isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  await connectToDatabase();
+  console.log("made it here");
+  const organization = await Organization.findOne({
+    where: { id: orgId },
+    relations: ['openSourceProjects']
+  });
+  console.log("my org = ", organization);
+  if (organization) {
+    return {
+      statusCode: 200,
+      body: JSON.stringify(organization.openSourceProjects)
+    };
+  }
+  return NotFound;
+});
+
+
+
+/**
+ * @swagger
+ *
+ * /project_upsert/org/{orgId}:
+ *  create:
+ *    description: Create a new open source project without an organization if it doesn't already exist and add it to a given organization.
+ *    tags:
+ *    - Organizations
+ */
+export const create_proj = wrapHandler(async (event) => {
+  const orgId = event.pathParameters?.orgId;
+  const parsedBody = JSON.parse(event.body ?? '{}');
+  const { url, hipcheckResults, organizations } = parsedBody;
+
+  // check permissions
+  if (!orgId || (!isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(orgId))) return Unauthorized;
+  
+  // const body = await validateBody(OpenSourceProject, event.body);
+  await connectToDatabase();
+
+  let openSourceProject: OpenSourceProject | undefined;
+
+  // Check if there's an existing open source project without an organization
+  const existingProjects = await OpenSourceProject.createQueryBuilder('osp')
+    .leftJoinAndSelect('osp.organizations', 'organizations')
+    .where('organizations.id IS NULL')
+    .getMany();
+
+  if (existingProjects.length > 0) {
+    // Use the first found project without organization
+    openSourceProject = existingProjects[0];
+  } else {
+    // Create a new open source project
+    openSourceProject = await OpenSourceProject.create({
+      // Set other properties as needed
+      url: url,
+      hipcheckResults: hipcheckResults    
+    });
+    await openSourceProject.save();
+  }
+
+  if (!openSourceProject.organizations) {
+    openSourceProject.organizations = [];
+  }
+
+  // Associate the open source project with the specified organization
+  const organization = await Organization.findOneOrFail(orgId);
+  if (!openSourceProject.organizations.find(org => org.id === organization.id)) {
+    openSourceProject.organizations.push(organization);
```

Comment:
I think (but could be wrong) that we can supply it with an object of {"id": orgId} rather than needing to find the organization.